packages/node_modules/node-red/red.js

```diff
@@ -142,208 +142,278 @@ if (process.env.NODE_RED_ENABLE_PROJECTS) {
     settings.editorTheme.projects.enabled = !/^false$/i.test(process.env.NODE_RED_ENABLE_PROJECTS);
 }
 
+var httpsPromise;
 if (settings.https) {
-    server = https.createServer(settings.https,function(req,res) {app(req,res);});
-} else {
-    server = http.createServer(function(req,res) {app(req,res);});
-}
-server.setMaxListeners(0);
+    var startupHttps = settings.https;
 
-function formatRoot(root) {
-    if (root[0] != "/") {
-        root = "/" + root;
-    }
-    if (root.slice(-1) != "/") {
-        root = root + "/";
+    if (typeof startupHttps === "function") {
+        // Get the result of the function, because createServer doesn't accept functions as input
+        startupHttps = startupHttps(); 
+    }  
+
+    if (startupHttps.hasOwnProperty('then') && typeof startupHttps.then === 'function') {
+        // A promise was returned
+        httpsPromise = startupHttps;
+    } else {
+        // An object was returned - wrap in a promise
+        httpsPromise = Promise.resolve(startupHttps);
     }
-    return root;
 }
-
-if (settings.httpRoot === false) {
-    settings.httpAdminRoot = false;
-    settings.httpNodeRoot = false;
-} else {
-    settings.httpRoot = settings.httpRoot||"/";
-    settings.disableEditor = settings.disableEditor||false;
+else {
+    // No https is enable - wrap null
+    httpsPromise = Promise.resolve(null);
 }
+    
+httpsPromise.then(function(startupHttps) {
+    if (startupHttps) {
+        server = https.createServer(startupHttps,function(req,res) {app(req,res);});
+    
+        // Refresh https settings at intervals for NodeJs version 11 and above
+        if (settings.httpsRefreshInterval) {
+            if (typeof settings.https === "function") {        
+                if (server.setSecureContext) {
+                    console.log("Refreshing https settings every " + parseInt(settings.httpsRefreshInterval) + " seconds.");
```

Comment:
This should be a properly translated message. As you have identified, the `RED` object hasn't been initialised yet, so the message catalog isn't available.

I suggest:
 - create a boolean flag that indicates if the refresh is enabled or not. Set it to true here, but set it back to false if any of the error conditions are hit.
 - later on, immediately before the `server.listen` call, you can then check the flag and log the message using  `RED.log.info(RED.log._("server.https.XYZ"))` with an appropriate message added to the catalog.
 - similarly, if any error was hit, they could be logged at this point as well.

Reply:
I have taken the liberty to do it a bit different:
1. I declare an array of log items:
   ```
   // Delay logging of (translated) messages until the RED object has been initialized
   var delayedLogItems = [];
   ```
2. I push at several places log items into this array (which can be error, warn or log), with optional parameters:
   ```
   delayedLogItems.push({type:"info", id:"server.https.refresh-interval", params:{interval:parseInt(settings.httpsRefreshInterval)}});
   ```
3. And once RED is initialized, I call (dynamically) the correct function for logging:
   ```
   // Log all the delayed messages, since they can be translated at this point
   delayedLogItems.forEach(function (delayedLogItem, index) {
      RED.log[delayedLogItem.type](RED.log._(delayedLogItem.id, delayedLogItem.params));
   });
   ```

I 'think' this is easier to reuse (compared to a checkbox-based solution), if you want to translate other log texts in the red.js file in the future...

Hope this is ok for you?